language-service/src/parser/jsonParser.ts

```diff
@@ -771,19 +773,36 @@ export class ObjectASTNode extends ASTNode {
 			});
 		}
 
-		let propertyProcessed = (prop: string) => {
+		const propertyProcessed = (prop: string): void => {
 			let index = unprocessedProperties.indexOf(prop);
 			while (index >= 0) {
 				unprocessedProperties.splice(index, 1);
 				index = unprocessedProperties.indexOf(prop);
 			}
 		};
 
+		const firstPropertyContains = (propertyName: string): boolean => {
+			return schema.firstProperty && schema.firstProperty.indexOf(propertyName) >= 0;
+		}
+
+		const isFirstProperty = (propertySchema: JSONSchema, propertyName: string): boolean => {
+			if (!schema.firstProperty || !schema.firstProperty.length) {
+				return false;
+			}
+
+			if (this.getIgnoreKeyCase(propertySchema)) {
+				const upperPropName: string = propertyName.toUpperCase();
+				return !!schema.firstProperty.find(listProperty => listProperty.toUpperCase() === upperPropName);
```

Comment:
I recently fixed a bug with `find` as it was not supported on Chrome 39 :) [find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) and [findIndex](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) are not supported in some older browsers and IE 11 so it is better to use [some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)